### PR TITLE
Xm registrar fixes

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

File: main/build/MacOSX/Makefile.am [hunk 1]
```diff
@@ -10,6 +10,8 @@ MD_CONFIGURE=$(top_srcdir)/../scripts/configure.sh
 EXTERNAL=../../external
 MONOSTUB_DEFINES=-DXM_REGISTRAR
 MONOSTUB_STATIC_LINK=$(EXTERNAL)/Xamarin.Mac.registrar.full.a
+HAS_EXTERN_C_XM_REGISTRAR_DEFINE=$(shell ./check-xm-extern.sh)
```
Comment (Therzok) [inline main/build/MacOSX/Makefile.am]: Self: remove HAS

File: main/build/MacOSX/check-xm-extern.sh [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#/usr/bin/env bash
+(nm ../../external/Xamarin.Mac.registrar.full.a | grep " _xamarin_create_classes_Xamarin_Mac" >/dev/null) && echo "-DEXTERN_C"
```
Comment (Therzok) [inline main/build/MacOSX/check-xm-extern.sh]: Self: add comment explaining

**Comment by slluis** [comment]
Can you explain what are the changes for?

**Comment by Therzok** [comment]
@slluis external/Makefile changes are so we now copy each file when it changes, not depend on Xamarin.Mac.dll being changed.

The other change is a check for specific versions of Xamarin.Mac. Newer Xamarin.Mac versions (master and I think d15-3) export the xamarin_create_classes function as "C", aka the compiler doesn't mangle the function name. This is needed so we can link the right function.

**Comment by Therzok** [comment]
Updated commits with better explanations. Merging.

**Pull request merged**